buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -258,8 +258,9 @@ private AdaptiveByteBuf allocateFallback(int size, int maxCapacity, Thread curre
         Magazine magazine;
         if (buf != null) {
             Chunk chunk = buf.chunk;
-            magazine = chunk != null && chunk != Magazine.MAGAZINE_FREED?
-                    chunk.currentMagazine() : getFallbackMagazine(currentThread);
+            if (chunk == null || chunk == Magazine.MAGAZINE_FREED || (magazine = chunk.currentMagazine()) == null) {
```

Comment:
I think `chunk == Magazine.MAGAZINE_FREED` can be removed, because the `MAGAZINE_FREED`'s `magazine` field is always null, and logically, the `buf.chunk` can never be `MAGAZINE_FREED`.

Reply:
It's also fine to keep it there, for insurance.